### PR TITLE
Add the plumbing for APIGW JWT work

### PR DESCRIPTION
**Pull request by jm96441n**
### Description

Adds the plumbing to hook up the JWT validation for APIGW

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [X] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: agent/consul/gateways/controller_gateways.go [hunk 1]
```diff
@@ -897,6 +923,31 @@ func invalidCertificates() structs.Condition {
 	)
 }
 
+// invalidJWTProvider returns a condition used when a gateway listener referecnes
```
Comment (missylbytes) [inline agent/consul/gateways/controller_gateways.go:926]: nit: references typo

File: agent/consul/gateways/controller_gateways.go [hunk 2]
```diff
@@ -944,6 +995,18 @@ func gatewayNotFound(ref structs.ResourceReference) structs.Condition {
 	)
 }
 
+// jwtProvderNotFound marks a Route as having failed to bind to a referenced APIGateway due to
```
Comment (missylbytes) [inline agent/consul/gateways/controller_gateways.go:998]: nit: typo jwtProvderNotFound

File: agent/consul/gateways/controller_gateways_ce.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	"github.com/hashicorp/consul/agent/consul/state"
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+func (r *apiGatewayReconciler) enqueueJWTProviderReferencedGatewaysAndHTTPRoutes(_ *state.Store, _ context.Context, _ controller.Request) error {
+	return nil
+}
+
+func (m *gatewayMeta) checkJWTProviders(_ *state.Store) (map[structs.ResourceReference]error, error) {
+	return nil, nil
```
Comment (sarahalsmiller) [inline agent/consul/gateways/controller_gateways_ce.go:22]: should this return a not allowed error in consul core, or is that future work

**Review by sarahalsmiller — APPROVED** [review]
Looks good to me, one question

File: agent/consul/gateways/controller_gateways_ce.go (hunk 3 above)
Comment (jm96441n) [reply agent/consul/gateways/controller_gateways_ce.go:22]: this probably shouldn't return anything since we're not actually doing any checks

**Pull request merged**